homeassistant/components/climacell/const.py

```diff
@@ -155,165 +150,184 @@
 CC_ATTR_CLOUD_BASE = "cloudBase"
 CC_ATTR_CLOUD_CEILING = "cloudCeiling"
 
-CC_SENSOR_TYPES = [
-    {
-        ATTR_FIELD: CC_ATTR_FEELS_LIKE,
-        ATTR_NAME: "Feels Like",
-        CONF_UNIT_SYSTEM_IMPERIAL: TEMP_FAHRENHEIT,
-        CONF_UNIT_SYSTEM_METRIC: TEMP_CELSIUS,
-        ATTR_METRIC_CONVERSION: lambda val: temp_convert(
-            val, TEMP_FAHRENHEIT, TEMP_CELSIUS
-        ),
-        ATTR_IS_METRIC_CHECK: True,
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-    },
-    {
-        ATTR_FIELD: CC_ATTR_DEW_POINT,
-        ATTR_NAME: "Dew Point",
-        CONF_UNIT_SYSTEM_IMPERIAL: TEMP_FAHRENHEIT,
-        CONF_UNIT_SYSTEM_METRIC: TEMP_CELSIUS,
-        ATTR_METRIC_CONVERSION: lambda val: temp_convert(
-            val, TEMP_FAHRENHEIT, TEMP_CELSIUS
-        ),
-        ATTR_IS_METRIC_CHECK: True,
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-    },
-    {
-        ATTR_FIELD: CC_ATTR_PRESSURE_SURFACE_LEVEL,
-        ATTR_NAME: "Pressure (Surface Level)",
-        CONF_UNIT_SYSTEM_IMPERIAL: PRESSURE_INHG,
-        CONF_UNIT_SYSTEM_METRIC: PRESSURE_HPA,
-        ATTR_METRIC_CONVERSION: lambda val: pressure_convert(
+
+@dataclass
+class ClimaCellSensorMetadata:
+    """Metadata about an individual ClimaCell sensor."""
+
+    field: str
+    name: str
+    unit_imperial: UnitT | None = None
+    unit_metric: UnitT | None = None
+    metric_conversion: Callable | float = 1.0
+    is_metric_check: bool | None = None
+    device_class: str | None = None
+    value_map: IntEnum | None = None
+
+    def __post_init__(self) -> None:
+        """Post initialization."""
+        if (self.unit_imperial or self.unit_metric) is not None and None in (
+            self.unit_imperial,
+            self.unit_metric,
+        ):
```

Comment:
```suggestion
        units = (self.unit_imperial, self.unit_metric)
        if any(u is not None for u in units) and any(u is None for u in units):
```
Might be a bit over-engineered, but what do you think?

Reply:
I think this is more clear in its intent, thanks for the suggestion!